src/inspector_js_api.cc

```diff
@@ -123,10 +124,7 @@ class JSBindingsConnection : public AsyncWrap {
     new JSBindingsConnection(env, info.This(), callback);
   }
 
-  void Disconnect() {
-    session_.reset();
-    delete this;
-  }
+  void Disconnect() { session_.reset(); }
```

Comment:
Is it an option to properly handle this here? That should fix the crash that’s being referred to in the PR description while also being clearly more correct.

```suggestion
  void Disconnect() {
    BaseObjectPtr<JSBindingsConnection> strong_ref{this};
    session_.reset(); 
    Detach();
  }
```

(This is a pattern we follow elsewhere as well, e.g. `StreamReq::Dispose()`, `MakeLibuvRequestCallback::Wrapper()`, `QueryWrap::QueueResponseCallback()`.)